### ISSUE TITLE
CalendarList disabled wrong days

### ISSUE THREAD

**Issue by anija** [wix/react-native-calendars#2086]
## Description

I'm using CalendarList with these params:
`current=2022-11-10`, `minDate=2022-11-10`and `maxDate=2022-11-17`.

I got a correct November calendar, but when I scroll down, each next month has all days disabled except the 10, that is highlighted like it is the current day.

If I remove the `current` prop, i got the same behaviour with 9th days (today is 09/11).

## Expected Behavior

Next months should be fully disabled.

## Observed Behavior

Each month after november has all days disabled except the 10.

## Environment

* `npm ls react-native-calendars`: react-native-calendars@1.1291.0
* `npm ls react-native`: react-native@0.68.3
* Simulator iPhone 13 iOS 15.5

## Reproducible Demo

Please provide a minimized reproducible demonstration of the problem you're reporting.

```
<CalendarList
  pastScrollRange={0}
  futureScrollRange={3}
  current={'2022-11-10'}
  hideArrows={true}
  hideExtraDays={true}
  enableSwipeMonths={false}
  minDate={'2022-11-10'}
  maxDate={'2022-11-17'}
  style={styles.calendar}
/>
```

## Screenshots

![Simulator Screen Shot - iPhone 13 - 2022-11-09 at 10 53 33](https://user-images.githubusercontent.com/2065066/200798919-3e1003e3-2b3c-4f8a-b8fc-6e0ec3fadae7.png)

**Comment by anija** [comment]
If anynone is intersted (certainly not the maintainers of the project), the bug is here:

/src/day-state-manager.js

The disabled status is not applied if the day is selected because of the `else if`. So the calendar list, thant mount each month passing "current" a day of the next month, wrongly set the day as `selected` instead of `disabled` because is out of min/maxDate range.

The ratio between the patches I've had to apply to this library and the answers I got on this page is unbelievable.
